usersync/cookie_test.go

```diff
@@ -262,51 +262,32 @@ func TestGetUIDsWithNilCookie(t *testing.T) {
 	assert.Len(t, uids, 0, "GetUIDs shouldn't return any user syncs for a nil cookie")
 }
 
-func TestUnlimitedSizeCookie(t *testing.T) {
-	var cookie *PBSCookie
-	var bigCookieLen int
-	cookie, bigCookieLen = newBigCookie()
-	//cookie.maxSizeBytes = 0 //When equal to zero, unlimited size
-
-	var received *PBSCookie = writeThenRead(cookie, 0)
-
-	assert.Equal(t, bigCookieLen, len(received.uids), "Cookie bigger than 32 KB in size was not supposed to be reduced in size")
-}
-
-func TestTrimBigCookie(t *testing.T) {
-	var cookie *PBSCookie
-	var bigCookieLen int
-	var maxCookieSize int = 1 << 15 // 32768 bytes = 32 KB
-	cookie, bigCookieLen = newBigCookie()
-
-	var received *PBSCookie = writeThenRead(cookie, maxCookieSize)
-
-	assert.Equal(t, bigCookieLen > len(received.uids), true, "Cookie bigger than 32 KB in size was not reduced according to date")
-}
-
-func TestDeleteClosestExpirationFromCookie(t *testing.T) {
+func TestTrimCookiesClosestExpirationDates(t *testing.T) {
 	cookieToSend, cookieToSendLen := newTestCookie()
+	closestToExpirationDate := "key7"
+
 	type aTest struct {
 		maxCookieSize int
 		expAction     string
 	}
 	testCases := []aTest{
 		{maxCookieSize: 2000, expAction: "equal"}, //1 don't trim, set
 		{maxCookieSize: 0, expAction: "equal"},    //2 unlimited size: don't trim, set
-		{maxCookieSize: 500, expAction: "trim"},   //3 trim to size and set
-		{maxCookieSize: 200, expAction: "empty"},  //4 insufficient size, trim to zero lenght and set
-		{maxCookieSize: -100, expAction: "empty"}, //5 invalid size, trim to zero lenght and set
+		{maxCookieSize: 800, expAction: "trim"},   //3 trim to size and set
+		{maxCookieSize: 500, expAction: "trim"},   //4 trim to size and set
+		{maxCookieSize: 200, expAction: "empty"},  //5 insufficient size, trim to zero lenght and set
+		{maxCookieSize: -100, expAction: "empty"}, //6 invalid size, trim to zero lenght and set
 	}
 	for i := range testCases {
 		processedCookie := writeThenRead(cookieToSend, testCases[i].maxCookieSize)
 		switch testCases[i].expAction {
 		case "equal":
-			assert.Equal(t, cookieToSendLen, len(processedCookie.uids), "[Test %d] MaxCookieSizeBytes equal to zero or bigger than %d bytes should be snough to set and remain cookie unchanged \n", i+1, len(processedCookie.uids))
-			_, oldestFound := processedCookie.uids["key7"]
+			assert.Equal(t, cookieToSendLen, len(processedCookie.uids), "[Test %d] MaxCookieSizeBytes equal to zero or bigger than %d bytes should be enough to set and remain cookie unchanged \n", i+1, len(processedCookie.uids))
+			_, oldestFound := processedCookie.uids[closestToExpirationDate]
 			assert.Equal(t, oldestFound, true, "[Test %d] Oldest entry in cookie should not have been eliminated", i+1)
```

Comment:
You can just do `assert.Contains` here on the map and skip the previous line.

Reply:
Done

Reply:
Done